src/integrations/GoogleCloudPubSubIntegration.ts

```diff
@@ -0,0 +1,336 @@
+            googleCloudPubSubIntegration.moduleNames, googleCloudPubSubIntegration.moduleVersion,
+            (lib: any, cfg: any, moduleName: string) => {
+                this.wrap.call(this, lib, cfg, moduleName);
+            },
+            (lib: any, cfg: any, moduleName: string) => {
+                this.doUnwrap.call(this, lib, moduleName);
+            },
+            this.config);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    wrap(lib: any, thundraConfig: any, moduleName: string): void {
+        ThundraLogger.debug('<GoogleCloudPubSubIntegration> Wrap');
+
+        const integration = this;
+        function requestWrapper(original: any, wrappedFunctionName: string) {
+            integration.wrappedFuncs[wrappedFunctionName] = original;
+            return function internalWrapper(config: any, callback: any) {
+
+                const orginalCallback = callback;
+                let span: ThundraSpan;
+                let reachedToCallOriginalFunc: boolean = false;
+                try {
+
+                    ThundraLogger.debug('<GoogleCloudPubSubIntegration> Tracing GoogleCloudPubSubIntegration request:', config);
+
+                    const { tracer } = ExecutionContextManager.get();
+
+                    if (!tracer) {
+                        ThundraLogger.debug('<GoogleCloudPubSubIntegration> Skipped tracing request as no tracer is available');
+                        return original.apply(this, [config, orginalCallback]);
+                    }
+
+                    const originalFunction = integration.getOriginalFunction(wrappedFunctionName);
+                    const parentSpan = tracer.getActiveSpan();
+
+                    const topic = GooglePubSubUtils.getTopic(config);
+                    const operationName = `${config.method.toUpperCase()}-${topic}`;
+
+                    ThundraLogger.debug(`<GoogleCloudPubSubIntegration> Starting publish span with name ${operationName}`);
+
+                    span = tracer._startSpan(operationName, {
+                        childOf: parentSpan,
+                        domainName: DomainNames.API,
+                        className: ClassNames.GOOGLEPUBSUB,
+                        disableActiveStart: true,
+                    });
+
+                    const getMessageBody = () => {
+                        if (!config || !config.reqOpts || !config.reqOpts.messages) {
+                            return;
+                        }
+
+                        return GooglePubSubUtils.parseMessages(config.reqOpts.messages);
+                    };
+
+                    const meesageContent = getMessageBody();
+
+                    const tags = {
+                        [GooglePubSubTags.PROJECT_ID]: this.projectId,
+                        [GooglePubSubTags.TOPIC_NAME]: topic,
+                        ...((meesageContent && !thundraConfig.maskGooglePublishRequestBody) ? {
+                            [GooglePubSubTags.MESSAGE]: JSON.stringify(getMessageBody()),
```

Comment:
What if we directly assign `getMessageBody()` without serializing it here as it will be already serialized while reporting?